src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_COEFFICIENT = 0.75f;
+    private static final int DEFAULT_RESIZE_COEFFICIENT = 2;
+    private Node<K, V>[] buckets;
+    private int currentLoad;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_INITIAL_CAPACITY];
+        currentLoad = (int) DEFAULT_LOAD_COEFFICIENT * DEFAULT_INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int hash = getHashCode(key);
+        int index = getIndex(key);
+        Node<K, V> newElement = new Node<>(hash, key, value, null);
+        Node<K, V> element = buckets[index];
+        if (element == null) {
+            buckets[index] = newElement;
+        }
+        while (element != null) {
+            if (Objects.equals(key, element.key)) {
+                element.value = value;
+                return;
+            }
+            if (element.next == null) {
+                element.next = newElement;
+                break;
+            }
+            element = element.next;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getHashCode(key);
+        int index = getIndex(key);
+        Node<K, V> element = buckets[index];
+        while (element != null) {
+            if (element.hash == hash && Objects.equals(key, element.key)) {
```

Comment:
Here is the same problem that Kostya tell you earlier. You don`t need hash check